backend-server/src/main/java/org/dice/ida/util/RDFUtil.java

```diff
@@ -366,4 +370,92 @@ public String addDatasetName(String dsName) {
 			return "false";
 		}
 	}
+
+	public Map<String, String> getParamDisplayNames() {
+		if(paramDisplayNameMap != null) {
+			return paramDisplayNameMap;
+		}
+		paramDisplayNameMap = new HashMap<>();
+		String queryString = IDAConst.IDA_SPARQL_PREFIX +
+				"SELECT ?paramLabel ?displayName " +
+				"WHERE { " +
+				"    ?s rdf:type ivoc:Parameter ; " +
+				"       rdfs:label ?paramLabel ; " +
+				"       ivodp:hasDisplayName ?displayName " +
+				"}";
+		ResultSet attributeResultSet = getResultFromQuery(queryString, "ida_viz");
+		if (attributeResultSet == null) {
+			return null;
+		}
+		while (attributeResultSet.hasNext()) {
+			QuerySolution querySolution = attributeResultSet.next();
+			String param = querySolution.get("paramLabel").asLiteral().getString();
+			String displayName = querySolution.get("displayName").asLiteral().getString();
+			paramDisplayNameMap.put(param, displayName);
+		}
+		if (conn != null) {
+			conn.close();
+		}
+		model = null;
+		return paramDisplayNameMap;
+	}
+
+	public Map<String, String> getParamDisplayMessages() {
+		if(paramDisplayMessageMap != null) {
+			return paramDisplayMessageMap;
+		}
+		paramDisplayMessageMap = new HashMap<>();
+		String queryString = IDAConst.IDA_SPARQL_PREFIX +
+				"SELECT ?paramLabel ?displayMessage " +
+				"WHERE { " +
+				"    ?s rdf:type ivoc:Parameter ; " +
+				"       rdfs:label ?paramLabel ; " +
+				"       ivodp:hasDisplayMessage ?displayMessage " +
+				"}";
```

Comment:
Better to use StringBuilder here. @nandeeshp24 could you please do it for all the similar scenarios?

Reply:
taken care of